townhall/myapi/services.py

```diff
@@ -21,9 +22,17 @@ def delete_volunteer(id: int) -> None:
         volunteer_dao.delete_volunteer(volunteer_id=id)
 
 class OpportunityServices:
-
     def get_opportunity(id: int) -> Opportunity:
-        return opportunity_dao.get_opportunity(id=id)
+        print(f"Fetching opportunity with ID: {id}")
+        opportunity = opportunity_dao.get_opportunity(id=id)
+        print(f"Fetched opportunity: {opportunity}")
+        return opportunity
+    
+    def get_opportunity_all() -> List[Opportunity]:
+        print("Fetching opportunities")
+        opportunities = Opportunity.objects.all()
```

Comment:
I think this should call the DAO method, service methods should never invoke DB operations like all() or get() or filter() etc